invisible_cities/cities/detsim_test.py

```diff
@@ -0,0 +1,277 @@
+
+
+def test_detsim_exact(ICDATADIR, output_tmpdir):
+
+    PATH_IN   = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT  = os.path.join(output_tmpdir, "detsim_test.h5")
+    PATH_TRUE = os.path.join(ICDATADIR    , "detsim_new_kr83m_fast.newformat.sim.h5")
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+
+    buffer_params = dict()
+    buffer_params["length"]      = 800 * units.mus
+    buffer_params["pmt_width"]   = 100 * units.ns
+    buffer_params["sipm_width"]  =   1 * units.mus
+    buffer_params["max_time"]    =  10 * units.ms
+    buffer_params["pre_trigger"] = 100 * units.mus
+    buffer_params["trigger_thr"] = 0
+
+    conf.update(dict(files_in      = PATH_IN,
+                     file_out      = PATH_OUT,
+                     run_number    = 0,
+                     event_range   = all_events,
+                     buffer_params = buffer_params))
+    np.random.seed(1234)
+    result = detsim(**conf)
+
+    tables = ["pmtrd", "sipmrd",
+              "Run/eventMap", "Run/events", "Run/runInfo",
+              "MC/configuration", "MC/event_mapping", "MC/hits",
+              "MC/particles", "MC/sns_positions", "MC/sns_response",
+              "Filters/active_hits"]
+
+    with tb.open_file(PATH_TRUE) as true_output_file:
+        with tb.open_file(PATH_OUT) as output_file:
+            for table in tables:
+                assert hasattr(output_file.root, table)
+                got      = getattr(     output_file.root, table)
+                expected = getattr(true_output_file.root, table)
+                assert_tables_equality(got, expected)
+
+
+def test_detsim_exact_time_translation(ICDATADIR, output_tmpdir):
+
+    PATH_IN   = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT  = os.path.join(output_tmpdir, "detsim_test.h5")
+
+    physics_params = dict()
+    physics_params["ws"]                     = 39.2 * units.eV
+    physics_params["wi"]                     = 22.4 * units.eV
+    physics_params["fano_factor"]            = 0.15
+    physics_params["conde_policarpo_factor"] = 1.00
+    physics_params["drift_velocity"]         = 1.00 * units.mm / units.mus
+    physics_params["lifetime"]               = 12   * units.ms
+    physics_params["transverse_diffusion"]   = 1.20 * units.mm / units.cm**0.5
+    physics_params["longitudinal_diffusion"] = 0.30 * units.mm / units.cm**0.5
+    physics_params["el_gain"]                = 365
+    physics_params["el_drift_velocity"]      = 2.5  * units.mm / units.mus
+
+    buffer_params = dict()
+    buffer_params["length"]      = 800 * units.mus
+    buffer_params["pmt_width"]   = 100 * units.ns
+    buffer_params["sipm_width"]  =   1 * units.mus
+    buffer_params["max_time"]    =  10 * units.ms
+    buffer_params["pre_trigger"] = 100 * units.mus
+    buffer_params["trigger_thr"] = 0
+
+    # copy to avoid mutability
+    physics_params_ = physics_params.copy()
```

Comment:
I dont see that you later plan to modify `physics_params` so why copying it again?

Reply:
`physics_params_` is mutted inside detsim city (`ws` and `el_drift_velocity`), therefore the second city call can not be done with the same dictionary. They will take the values from **detsim.conf** from now on.